lib/miq_automation_engine/service_models/miq_ae_service_host.rb

```diff
@@ -65,14 +59,7 @@ def ems_custom_get(key)
     end
 
     def ems_custom_set(attribute, value)
-      MiqQueue.put(
-        :class_name  => @object.class.name,
-        :instance_id => @object.id,
-        :method_name => 'set_custom_field',
-        :zone        => @object.my_zone,
-        :role        => 'ems_operations',
-        :args        => [attribute, value]
-      ) if @object.is_vmware?
+      sync_or_async_ems_operation(false, 'set_custom_field', [attribute, value]) if @object.is_vmware?
```

Comment:
This method is currently implemented in the base `Host` model and is checking for `is_vmware?` so there is no need to do the check in the service model. (See https://github.com/ManageIQ/manageiq/blob/87061966261086b377c85cf09302938ab7be3029/app/models/host.rb#L1264)

As a followup PR this should be changed to follow the implementation in `vm_or_template.rb` to use the `raw_*` pattern.

For now I would just suggest removing the `is_vmware?` check.